gk/main.c

```diff
@@ -417,21 +425,141 @@ setup_gk_instance(unsigned int lcore_id, struct gk_config *gk_conf)
 		goto flow_hash;
 	}
 
+	ret = snprintf(ring_name, sizeof(ring_name), "gk_ring_%d", block_idx);
```

Comment:
That's not a sound argument because Cody did not get the ARP cache merged yet, that is, that code is not peer reviewed.

I think it's worth the time to push the code into a library. Not only will it simplify reading the code, but it'll also save time of future pull requests that will need to use mailboxes.